google/auth/compute_engine/credentials.py

```diff
@@ -108,3 +117,163 @@ def service_account_email(self):
     def requires_scopes(self):
         """False: Compute Engine credentials can not be scoped."""
         return False
+
+
+class Signer(crypt.Signer):
+    """Signer that uses the default service account of a a GCE instance.
+
+    See the signBlob method in
+    https://cloud.google.com/iam/reference/rest/v1/projects.serviceAccounts
+    """
+
+    _REQUEST_PATH = (
+        "https://iam.googleapis.com/v1/"
+        "projects/{project}/serviceAccounts/{service_account}:signBlob")
+
+    def __init__(self):
+        self._creds = Credentials()
+        request = Request()
+        project_id = _metadata.get_project_id(request)
+        sa_info = _metadata.get_service_account_info(request)
+        self._service_account_email = sa_info['email']
+        self._request_path = self._REQUEST_PATH.format(
+            project=project_id,
+            service_account=self._service_account_email)
+        self._session = AuthorizedSession(self._creds)
+
+    @property
+    def key_id(self):
+        """Optional[str]: The key ID used to identify this private key.
+
+        There is no known key ID associated to the default service account.
+        Any sign() call may be signed with a different key.
+        """
+        return None
+
+    def sign(self, message):
+        """Signs a message.
+
+        Args:
+            message (Union[str, bytes]): The message to be signed.
+
+        Returns:
+            bytes: The signature of the message.
+        """
+        if not isinstance(message, bytes):
+            message = message.encode()
+        body = {
+            'bytesToSign': base64.b64encode(message).decode(),
+        }
+        rep = self._session.post(self._request_path, json=body)
+        rep.raise_for_status()
+        # Note: the response includes a key ID in rep.json()['keyId']
+        return base64.b64decode(rep.json()["signature"].encode())
+
+    @property
+    def service_account_email(self):
+        """The email of the default service account on this GCE instance."""
+        return self._service_account_email
+
+
+_DEFAULT_TOKEN_LIFETIME_SECS = 3600  # 1 hour in seconds
+_DEFAULT_TOKEN_URI = 'https://www.googleapis.com/oauth2/v4/token'
+
+
+class IDTokenCredentials(credentials.Credentials, credentials.Signing):
+    """Open ID Connect ID Token-based service account credentials.
+
+    These credentials relies on the default service account of a GCE instance.
```

Comment:
Could you get it to work with any of the service accounts on the GCE instance?

Reply:
Done